cmd/cli/cli.go

```diff
@@ -216,64 +216,102 @@ func sourceWriteFunc(t targetiface.Target, ft failureiface.Failure, tr transform
 		// Send message buffer
 		messagesToSend := transformed.Result
 
-		res, err := retry.ExponentialWithInterface(5, time.Second, "target.Write", func() (interface{}, error) {
-			res, err := t.Write(messagesToSend)
+		write := func() (*models.TargetWriteResult, error) {
+			result, err := t.Write(messagesToSend)
+
+			o.TargetWrite(result)
+			messagesToSend = result.Failed
+			return result, err
+		}
+
+		writeResult, err := handleWrite(cfg, write)
 
-			o.TargetWrite(res)
-			messagesToSend = res.Failed
-			return res, err
-		})
 		if err != nil {
 			return err
 		}
-		resCast := res.(*models.TargetWriteResult)
 
 		// Send oversized message buffer
-		messagesToSend = resCast.Oversized
+		messagesToSend = writeResult.Oversized
 		if len(messagesToSend) > 0 {
-			err2 := retry.Exponential(5, time.Second, "failureTarget.WriteOversized", func() error {
-				res, err := ft.WriteOversized(t.MaximumAllowedMessageSizeBytes(), messagesToSend)
-				if err != nil {
-					return err
-				}
-				if len(res.Oversized) != 0 || len(res.Invalid) != 0 {
+
+			writeOversized := func() (*models.TargetWriteResult, error) {
+				result, err := ft.WriteOversized(t.MaximumAllowedMessageSizeBytes(), messagesToSend)
+				if len(result.Oversized) != 0 || len(result.Invalid) != 0 {
 					log.Fatal("Oversized message transformation resulted in new oversized / invalid messages")
 				}
 
-				o.TargetWriteOversized(res)
-				messagesToSend = res.Failed
+				o.TargetWriteOversized(result)
+				messagesToSend = result.Failed
+				return result, err
+			}
+
+			_, err := handleWrite(cfg, writeOversized)
```

Comment:
Do we not do anything with the write result any more? If I'm not mistaken this would've been used for observer reporting, no?

Reply:
This is result of `handleWrite` function, so function which wraps umderlying target.write functions with retrying.

In this case we wrap [writeOversized](https://github.com/snowplow/snowbridge/blob/3089177fec596430e1932bb238607c07851d3142/cmd/cli/cli.go#L237) and in this function we indeed use result and pass them to [observer](https://github.com/snowplow/snowbridge/blob/3089177fec596430e1932bb238607c07851d3142/cmd/cli/cli.go#L243). So we use them but on different level (like previously)